### ISSUE TITLE
[MODEL DEFECT] All models fail SQA baseline - Insufficient TF-IDF vocabulary (max_features=50)

### ISSUE THREAD

**Issue by Ahmad-AbuGhazaleh** [Ahmad-AbuGhazaleh/DS-AI-SQA-Project#6]
## Issue Description
All three classification models fail to meet the SQA baseline accuracy of 70%. Root cause is severely limited TF-IDF vocabulary with only 50 features, which is insufficient for effective text classification.

## Current Performance

**All Models FAIL SQA Baseline:**

| Model | Test Accuracy | Precision | Recall | F1-Score | Status |
|-------|---------------|-----------|--------|----------|--------|
| Logistic Regression | 63.50% | 0.6327 | 0.3605 | 0.4593 | ❌ FAIL |
| Naive Bayes | 63.50% | 0.6757 | 0.2907 | 0.4065 | ❌ FAIL |
| Linear SVM | 61.50% | 0.5849 | 0.3605 | 0.4460 | ❌ FAIL |

**SQA Requirement:** Accuracy ≥ 70%  
**Best Achieved:** 63.50% (6.5 percentage points below baseline)

## Root Cause Analysis

**Problem:** TF-IDF vectorizer configured with `max_features=50`
```python
vectorizer = TfidfVectorizer(
    max_features=50,      # SEVERELY LIMITED
    ...
)
```

**Impact:**
- Vocabulary size: Only 50 words
- Expected for good performance: 1000-5000 words
- **Reduction:** 95-99% of important features are missing
- Models cannot learn meaningful patterns

**Why This Hurts Performance:**
1. **Insufficient Information:** 50 words cannot capture sentiment nuances
2. **Missing Key Terms:** Important sentiment words excluded from vocabulary
3. **Poor Recall:** Only 36% recall (missing 64% of positive sentiments)
4. **Weak F1-Scores:** All below 0.46 (poor balance of precision/recall)

## Expected Behavior
According to SQA Plan Section 2:
- "Classification Metrics (Sentiment140): Evaluated using Accuracy, Precision, Recall, and F1-Score"
- Models should achieve ≥70% accuracy baseline
- F1-Score should be balanced (>0.65)

## Comparison with Industry Standards

**Sentiment140 Benchmarks:**
- Baseline (random): 50%
- Basic models (proper features): 75-80%
- **Our current performance:** 61-63.5% (barely above random)

## Proposed Solution

Increase `max_features` to appropriate level:
```python
vectorizer = TfidfVectorizer(
    max_features=3000,    # Increased from 50
    min_df=2,
    max_df=0.8,
    ngram_range=(1, 2),   # Add bigrams for better context
    ...
)
```

**Expected Improvements:**
- Vocabulary: 50 → 3000 words (60x increase)
- Accuracy: 63% → 75-80%
- F1-Score: 0.45 → 0.72-0.78
- All models should pass 70% baseline

## Additional Optimizations

Consider also:
1. Adding bigrams (ngram_range=(1, 2)) for better context
2. Implementing K-Fold Cross-Validation per SQA Plan
3. Fine-tuning other hyperparameters

## Environment
- Dataset: Sentiment140 (1000 tweet sample)
- Notebook: sentiment140_analysis.ipynb
- Branch: feature/sentiment140-analysis
- Current max_features: 50
- Proposed max_features: 3000

---
**Severity:** High (All models fail SQA baseline)  
**Type:** Model Defect / Configuration Error  
**Reported by:** Ahmad Abu Ghazaleh  
**Student ID:** 202210221  
**Date:** December 27, 2025
